core/src/main/java/org/apache/iceberg/BaseMetadataTable.java

```diff
@@ -80,7 +80,7 @@ static PartitionSpec transformSpec(Schema metadataTableSchema, PartitionSpec spe
       int newFieldId = reassignedFields.getOrDefault(field.fieldId(), field.fieldId());
       builder.add(newFieldId, newFieldId, field.name(), Transforms.identity());
     }
-    return builder.build();
+    return builder.build(true);
```

Comment:
Please add a short comment to explain why we intentionally bypass the compatibility check here